rust/src/nasl/builtin/host/mod.rs

```diff
@@ -79,14 +133,186 @@ fn nasl_get_host_ip(
     Ok(NaslValue::String(ip.to_string()))
 }
 
+/// Get an IP address corresponding to the host name
+fn resolve_host_name(
+    register: &Register,
+    _context: &Context,
+) -> Result<NaslValue, FunctionErrorKind> {
+    let hostname = match register.named("hostname") {
+        Some(ContextType::Value(NaslValue::String(x))) if !x.is_empty() => x.clone(),
+        _ => {
+            return Err(FunctionErrorKind::diagnostic_ret_null("Missing Hostname"));
+        }
+    };
+
+    match resolve(hostname)? {
+        Some(mut a) => {
+            let address = a.next().map_or_else(String::new, |x| x.to_string());
+            let address = &address[..(address.len() - 5)];
```

Comment:
What is the 5? I think what youre trying to do is strip the port from the ip addr? This should be changed by returning `IpAddr` from `resolve` (which is the only thing that makes sense anyways, see below) and not by doing custom parsing logic on the `Display` result of `SocketAddr`